docs/ReferenceGuides/UnderscoredAttributes.md

```diff
@@ -0,0 +1,428 @@
+}
+
+func baz() {
+  let ptr = UnsafePointer(&global) // okay
+  let ptr2 = UnsafePointer(&S.staticVar) // okay
+}
+```
+
+Additionally, if they are of a tuple or struct type, their stored members
+without observers may also be passed inout as non-ephemeral pointers.
+
+For more details, see [the educational note on temporary pointer usage](/userdocs/diagnostics/temporary-pointers.md).
+
+## `@_nonoverride`
+
+## `@_objc_non_lazy_realization`
+
+## `@_optimize([none|size|speed])`
+
+Controls the compiler's optimization mode. This attribute is analogous to the
+command-line flags `-Onone`, `-Osize` and `-Ospeed` respectively, but limited
+to a single function body.
+
+`@_optimize(none)` is handy for diagnosing and reducing compiler bugs as well
+as improving debugging in Release builds.
+
+## `@_originallyDefinedIn(module: "ModuleName", availabilitySpec...)`
+
+Marks a declaration as being originally defined in a different module,
+changing the name mangling. This can be used to move declarations
+from a module to one of the modules it imports without breaking clients.
+
+Consider the following example where a framework ToasterKit needs
+to move some APIs to a lower-level framework ToasterKitCore.
+Here are the necessary changes:
+
+1. Add a linker flag `-reexport_framework ToasterKitCore` for ToasterKit.
+   This ensures all symbols defined in ToasterKitCore will be accessible during
+   runtime via ToasterKit, so existing apps continue to run.
+2. In ToasterKit, use `@_exported import ToasterKitCore`.
+   This ensures existing source code that only imports ToasterKit continues to
+   type-check.
+3. Move the necessary declarations from ToasterKit to ToasterKitCore.
+   The moved declaration should have two attributes:
+   - `@available` indicating when the declaration was introduced in ToasterKit.
+   - `@_originallyDefinedIn` indicating the original module and when the
+     declaration was moved to ToasterKitCore.
+   ```swift
+   @available(toasterOS 42, *)
+   @_originallyDefinedIn(module: "ToasterKit", toasterOS 57)
+   enum Toast {
+   case underdone
+   case perfect
+   case burnt
+   }
+   ```
+4. Add Swift compiler flags `-Xfrontend -emit-ldadd-cfile-path -Xfrontend /tmp/t.c`
+   to ToasterKitCore's build settings. Add the emitted `/tmp/t.c` file to
+   ToasterKit's compilation.
+   This ensures when an app is built for deployment targets prior to the symbols' move,
+   the app will look for these symbols in ToasterKit instead of ToasterKitCore.
+
+More generally, mutliple availabilities can be specified, like so:
+
+```swift
+@available(toasterOS 42, bowlOS 54, mugOS 54, *)
+@_originallyDefinedIn(module: "ToasterKit", toasterOS 57, bowlOS 69, mugOS 69)
+enum Toast { ... }
+```
+
+## `@_private(sourceFile: "FileName.swift")`
+
+Fully bypasses access control, allowing access to private declarations
+in the imported module. The imported module needs to be compiled with
+`-Xfrontend -enable-private-imports` for this to work.
```

Comment:
Inquiring minds would want to know, what was this designed for/where is it used?